### ISSUE TITLE
ParentNotInitializedException when setNoClassPath is true

### ISSUE THREAD

**Issue by ttben** [INRIA/spoon#1747]
I am trying to analyse the core code of an Android app.
The decompiled APK contains several packages and I want to consider only one of them.

```
Launcher spoon = new Launcher();
spoon.addInputResource("full/path/to/org");
spoon.buildModel(); <-- this crashes
```

> Exception in thread "main" spoon.compiler.ModelBuildingException: Callback cannot be resolved to a type at ...

It yields an error because it was looking for a type that does not exist in the classpath. I could add more input sources, but I wanted to "isolate" the source code instead of aggregating every source until it works.

Thus, I added the following statement just after the creation of the `Launcher` as the documentation described my situation [1]

`spoon.getEnvironment().setNoClasspath(true);`

> spoon.reflect.declaration.ParentNotInitializedException: parent not initialized for (class spoon.support.reflect.reference.CtExecutableReferenceImpl)

I don't understand this error. Is it refering to the `CtExecutableReferenceImp` itself, or an element of this type in the analyzed code? Do I have to set another flag? Is it an expected behavior?
The main terminated on a error status (code 1).

Thanks

[1]
> Sets the option "noclasspath", use with caution (see explanation below). With this option, Spoon does not require the full classpath to build the model. In this case, all references to classes that are not in the classpath are handled with the reference mechanism. The "simplename" of the reference object refers to the unbound identifier. This option facilitates the use of Spoon when is is hard to have the complete and correct classpath, for example for mining software repositories. For writing analyses, this option works well if you don't cross the reference by a call to getDeclaration() (if you really want to do so, then check for nullness of the result before). In normal mode, compilation errors are signaled as exception, with this option enabled they are signaled as message only. The reason is that in most cases, there are necessarily errors related to the missing classpath elements.

**Comment by ttben** [comment]
I tried with the `5.9.0` and the `6.0.0` and the error remains.
In `5.9.0` the logs ERROR dont appear, thats all.
```
2017-11-24 13:49:33,185 ERROR spoon.support.compiler.jdt.JDTTreeBuilder - Could not find declaration for variable ErrorUtils at (/Users/benjaminbenni/Downloads/SoundWavesPodcastPlayer_v0.158.8_apkpure.com(1)_source_from_JADX/org/bottiger/podcast/webservices/directories/audiosearch/AudioSearch.java:109)
2017-11-24 13:49:33,189 ERROR spoon.support.compiler.jdt.JDTTreeBuilder - Could not find declaration for variable ErrorUtils at (/Users/benjaminbenni/Downloads/SoundWavesPodcastPlayer_v0.158.8_apkpure.com(1)_source_from_JADX/org/bottiger/podcast/webservices/directories/audiosearch/AudioSearch.java:121)
2017-11-24 13:49:33,209 ERROR spoon.support.compiler.jdt.JDTTreeBuilder - Could not find declaration for variable ErrorUtils at (/Users/benjaminbenni/Downloads/SoundWavesPodcastPlayer_v0.158.8_apkpure.com(1)_source_from_JADX/org/bottiger/podcast/webservices/directories/itunes/ITunes.java:168)
2017-11-24 13:49:33,842 ERROR spoon.support.compiler.jdt.JDTTreeBuilder - Could not find declaration for variable VendorCrashReporter at (/Users/benjaminbenni/Downloads/SoundWavesPodcastPlayer_v0.158.8_apkpure.com(1)_source_from_JADX/org/bottiger/podcast/activities/feedview/FeedActivity.java:442)
2017-11-24 13:49:33,847 ERROR spoon.support.compiler.jdt.JDTTreeBuilder - Could not find declaration for variable Log at (/Users/benjaminbenni/Downloads/SoundWavesPodcastPlayer_v0.158.8_apkpure.com(1)_source_from_JADX/org/bottiger/podcast/activities/feedview/FeedActivity.java:443)
Exception in thread "main" spoon.reflect.declaration.ParentNotInitializedException: parent not initialized for (class spoon.support.reflect.reference.CtExecutableReferenceImpl)
	at spoon.support.reflect.declaration.CtElementImpl.getParent(CtElementImpl.java:322)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:29)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:157)
	at spoon.reflect.visitor.CtScanner.visitCtInvocation(CtScanner.java:495)
	at spoon.support.reflect.code.CtInvocationImpl.accept(CtInvocationImpl.java:56)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:157)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:141)
	at spoon.reflect.visitor.CtScanner.visitCtBlock(CtScanner.java:291)
	at spoon.support.reflect.code.CtBlockImpl.accept(CtBlockImpl.java:48)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:157)
	at spoon.reflect.visitor.CtScanner.visitCtConstructor(CtScanner.java:359)
	at spoon.support.reflect.declaration.CtConstructorImpl.accept(CtConstructorImpl.java:57)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:157)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:141)
	at spoon.reflect.visitor.CtScanner.visitCtClass(CtScanner.java:327)
	at spoon.support.reflect.declaration.CtClassImpl.accept(CtClassImpl.java:68)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:157)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:141)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:640)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:56)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:157)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:141)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:639)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:56)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:157)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:141)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:639)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:56)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:157)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:141)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:639)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:56)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.AstParentConsistencyChecker.scan(AstParentConsistencyChecker.java:36)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:157)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:141)
	at spoon.reflect.visitor.CtScanner.visitCtPackage(CtScanner.java:639)
	at spoon.support.reflect.declaration.CtPackageImpl.accept(CtPackageImpl.java:56)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.checkModel(JDTBasedSpoonCompiler.java:132)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:126)
	at spoon.support.compiler.jdt.JDTBasedSpoonCompiler.build(JDTBasedSpoonCompiler.java:100)
	at spoon.Launcher.buildModel(Launcher.java:709)
	at android.conflicts.CustomLauncher.main(CustomLauncher.java:46)
```

**Comment by ttben** [comment]
This class is the reason of the crash:
```
package org.bottiger.podcast.adapters;

import android.graphics.Bitmap;
import android.widget.ImageView;
import com.bumptech.glide.g.a.c;
import org.bottiger.podcast.adapters.viewholders.subscription.SubscriptionViewHolder;

class SubscriptionAdapter$3 extends c {
    final /* synthetic */ SubscriptionAdapter this$0;
    final /* synthetic */ SubscriptionViewHolder val$argHolder;

    SubscriptionAdapter$3(SubscriptionAdapter subscriptionAdapter, ImageView imageView, SubscriptionViewHolder subscriptionViewHolder) {
        this.this$0 = subscriptionAdapter;
        this.val$argHolder = subscriptionViewHolder;
        super(imageView);
    }

    protected void setResource(Bitmap bitmap) {
        this.val$argHolder.image.setImageBitmap(bitmap);
    }
}
```

Again, this is a code from a **decompiled APK**.
The `super(imageView)`  call crashes.

**Comment by monperrus** [comment]
> Again, this is a code from a *decompiled APK*.
Interesting. using which decompiler?

**Comment by ttben** [comment]
I used two online decompilers that use JADX

**Comment by monperrus** [comment]
closed per #1749
